File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {

        unordered_map<int, int> stack;
        
        for (int i = 0; i < nums.size(); i++) {
            int diff = target - nums[i];
     
            if (stack.find(diff) != stack.end()) {
                return {stack[diff], i};
            }
            
        
            stack[nums[i]] = i;
        }
        
        return {}; 
    }
};

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        twopass = {}
        
        for i in range(0,len(nums)):
            twopass[nums[i]] = i

        for i in range(0,len(nums)):
            diff = target - nums[i]
            if diff in twopass and twopass[diff] != i:
                return [i,twopass[diff]]
        
            
        
        return None

            
